integrations/qdrant/tests/test_retriever.py

```diff
@@ -112,3 +113,253 @@ def test_run(self, filterable_docs: List[Document]):
+                        "wait_result_from_api": True,
+                        "metadata": {},
+                        "write_batch_size": 100,
+                        "scroll_size": 10000,
+                        "payload_fields_to_index": None,
+                    },
+                },
+                "filters": None,
+                "top_k": 10,
+                "scale_score": True,
+                "return_embedding": False,
+            },
+        }
+
+    def test_from_dict(self):
+        data = {
+            "type": "haystack_integrations.components.retrievers.qdrant.retriever.QdrantSparseRetriever",
+            "init_parameters": {
+                "document_store": {
+                    "init_parameters": {"location": ":memory:", "index": "test"},
+                    "type": "haystack_integrations.document_stores.qdrant.document_store.QdrantDocumentStore",
+                },
+                "filters": None,
+                "top_k": 5,
+                "scale_score": False,
+                "return_embedding": True,
+            },
+        }
+        retriever = QdrantSparseRetriever.from_dict(data)
+        assert isinstance(retriever._document_store, QdrantDocumentStore)
+        assert retriever._document_store.index == "test"
+        assert retriever._filters is None
+        assert retriever._top_k == 5
+        assert retriever._scale_score is False
+        assert retriever._return_embedding is True
+
+    def _generate_mocked_sparse_embedding(self, n):
+        list_of_sparse_vectors = []
+        for _ in range(n):
+            random_indice_length = np.random.randint(3, 15)
+            data = {
+                "indices": list(range(random_indice_length)),
+                "values": [np.random.random_sample() for _ in range(random_indice_length)],
+            }
+            list_of_sparse_vectors.append(data)
+        return list_of_sparse_vectors
+
+    def test_run(self, filterable_docs: List[Document]):
+        document_store = QdrantDocumentStore(location=":memory:", index="Boi")
+
+        # Add fake sparse embedding to documents
+        for doc in filterable_docs:
+            doc.meta["_sparse_vector"] = self._generate_mocked_sparse_embedding(1)[0]
+
+        document_store.write_documents(filterable_docs)
+
+        retriever = QdrantSparseRetriever(document_store=document_store)
+
+        results: List[Document] = retriever.run(query_sparse_embedding=self._generate_mocked_sparse_embedding(1)[0])
+
+        assert len(results["documents"]) == 10  # type: ignore
+
+        results = retriever.run(
+            query_sparse_embedding=self._generate_mocked_sparse_embedding(1)[0], top_k=5, return_embedding=False
+        )
+
+        assert len(results["documents"]) == 5  # type: ignore
+
+        for document in results["documents"]:  # type: ignore
+            assert document.embedding is None
+
+
+# class TestQdrantHybridRetriever(FilterableDocsFixtureMixin):
```

Comment:
Commented because not really working. Search_Batch return ScoredPoint and not Point, so they are not convertible to Haystack documents (no payload)

Reply:
fixed